tests/test_task.py

```diff
@@ -0,0 +1,134 @@
+"""This module contains the tests for the task manager."""
+from multiprocessing.pool import AsyncResult
+
+import pytest
+
+from aea.skills.base import Task
+from aea.skills.tasks import TaskManager
+
+
+class MyTask(Task):
+    def __init__(self, return_value, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.setup_called = False
+        self.teardown_called = False
+        self.execute_called = False
+        self.execute_args = None
+        self.execute_kwargs = None
+        self.return_value = return_value
+
+    def setup(self) -> None:
+        self.setup_called = True
+
+    def execute(self, *args, **kwargs) -> None:
+        self.execute_called = True
+        self.execute_args = args
+        self.execute_kwargs = kwargs
+        return self.return_value
+
+    def teardown(self) -> None:
+        self.teardown_called = True
+
+
+class TestTaskManager:
+    """Test the features of the task manager."""
+
+    def _return_a_constant(self, a: int, b: int = 10):
+        return a + b
+
+    @classmethod
+    def setup_class(cls):
+        """Set the tests up."""
+        cls.task_manager = TaskManager(nb_workers=5)
+        cls.task_manager.start()
+
+    def test_task_manager_function_with_default_arguments(self):
+        """Test a function submitted to the task manager with default arguments."""
+        task_result = self.task_manager.enqueue_task(
+            self._return_a_constant, args=(32,)
+        )
+        assert isinstance(task_result, AsyncResult)
+        result = task_result.get(2.0)
+        assert result == 42
+
+    def test_task_manager_function_with_keyword_arguments(self):
+        """Test a function submitted to the task manager with keyword arguments."""
+        task_result = self.task_manager.enqueue_task(
+            self._return_a_constant, args=(32,), kwds={"b": 10}
+        )
+        assert isinstance(task_result, AsyncResult)
+        result = task_result.get(2.0)
+        assert result == 42
+
+    def test_task_manager_function_with_wrong_argument_number(self):
+        """Test wrong number of arguments."""
+        task_result = self.task_manager.enqueue_task(
+            self._return_a_constant, args=(), kwds={"b": 10}
+        )
+        assert isinstance(task_result, AsyncResult)
+        with pytest.raises(TypeError, match="missing .+ required positional argument:"):
+            task_result.get(2.0)
+
+    def test_task_manager_task_object(self):
+        """Test task manager with task object."""
+        expected_args = (0, 1)
+        expected_kwargs = {"a": 0, "b": 2}
+        expected_return_value = 42
+        my_task = MyTask(
+            return_value=expected_return_value, name="my_task", skill_context=object()
+        )
+        task_result = self.task_manager.enqueue_task(
+            my_task, args=expected_args, kwds=expected_kwargs
+        )
+        assert isinstance(task_result, AsyncResult)
+
+        expected_task = task_result.get(2.0)
```

Comment:
pay attention to this. `expected_task` is a *copy* of `my_task`. It might not be desirable in all cases.

BTW, worth mentioning that the result is indeed the `Task` copy because in the `Task.__call__` we return `self`.
Please see https://github.com/fetchai/agents-aea/pull/680/files#r372136135